src/smolagents/cli.py

```diff
@@ -76,10 +76,22 @@ def parse_arguments():
         type=str,
         help="The API key for the model",
     )
+    group.add_argument(
+        "--provider",
+        type=str,
+        default="hf-inference",
```

Comment:
better to default to `None` given we wil change the default provider in huggingface_hub in the future

Reply:
let's keep it like this and I'll change when it changes, does it work?

Reply:
hmm why not leave it as None?

Reply:
hmm I thought it'd break, I made the changes